vmm/src/vm.rs

```diff
@@ -908,6 +929,45 @@ impl Vm {
         }
     }
 
+    fn migration_request_handle(&self, receiver: Receiver<MigrationRequest>) -> Result <()> {
+        thread::Builder::new()
+            .name(format!("migration_register"))
+            .spawn(move || {
+                let request = receiver.recv();
+
+                match request {
+                    Ok(req) => {
+                        match req {
+                            MigrationRequest::MigrationStateGet(sender) => {
+                                /* TODO: assemble Vm states and send it as payload */
+                            }
```

Comment:
I don't think we should have to handle asynchronous requests from migratable components for getting a state, as explained in another comment.

Reply:
I do not quite understand your point. If you want to send payload as
response to 'MigrationRegister', this request must be sent by migration
thread to the component which holds states. And, the component should hold
the api receiver too. But there should be only one api receiver which is
held by 'control_loop()' and it does not know which component has states
to migrate. Only the component itself should register its sender to
migration thread.

So, I think the api sender should be input into the component to register
its migration sender to migration thread. Then, migration thread sends
another request to get states back.

If my understand is not right, please help to correct me. Thanks!